apollo-federation/src/type_merging.rs

```diff
@@ -0,0 +1,1364 @@
+                    examples.insert("Input", input_example.clone());
+                    examples.insert("Output", output_example.clone());
+                    examples
+                }
+                Some(EnumTypeUsage::Unused) => HashMap::new(),
+                None => HashMap::new(),
+            };
+
+            // Add example for current position if not already present
+            if !examples.contains_key(this_position) {
+                // Find first non-null source
+                let mut example_coordinate = element_name.to_string();
+                for (idx, source_type) in type_sources_helpers::iter_types(sources) {
+                    if source_type.is_some() {
+                        // Use the coordinate from the first available source
+                        // We need to get the subgraph name from the index
+                        if let Some(subgraph_name) = self.subgraph_names.get(idx) {
+                            example_coordinate = format!("{}@{}", element_name, subgraph_name);
+                            break;
+                        }
+                    }
+                }
+                examples.insert(this_position, example_coordinate);
+            }
+
+            // Convert back to EnumTypeUsage format
+            let new_usage = match position {
+                "Input" => EnumTypeUsage::Input {
+                    input_example: examples
+                        .get("Input")
+                        .unwrap_or(&element_name.to_string())
+                        .clone(),
+                },
+                "Output" => EnumTypeUsage::Output {
+                    output_example: examples
+                        .get("Output")
+                        .unwrap_or(&element_name.to_string())
+                        .clone(),
+                },
+                "Both" => EnumTypeUsage::Both {
+                    input_example: examples
+                        .get("Input")
+                        .unwrap_or(&element_name.to_string())
+                        .clone(),
+                    output_example: examples
+                        .get("Output")
+                        .unwrap_or(&element_name.to_string())
+                        .clone(),
+                },
+                _ => unreachable!("Invalid position"),
+            };
+
+            // Store the updated usage
+            self.enum_usages.insert(enum_name, new_usage);
+        }
+    }
+
+    /// Extract the base type name from a type (removing nullability and list wrappers)
+    fn get_base_type(&self, typ: &Type) -> String {
+        get_base_type_impl(typ)
+    }
+
+    /// Check if two types are the same (including nullability and list structure) -  sameType
+    fn same_type(&self, type1: &Type, type2: &Type) -> bool {
+        same_type_impl(type1, type2)
+    }
+
+    /// Check if type1 is a strict subtype of type2
+    fn is_strict_subtype(&self, type1: &Type, type2: &Type) -> Result<bool, FederationError> {
```

Comment:
Rather than type1 and type2, it would be good for the name to indicate which is the type being checked and which being checked against. Comment for all these type helper functions